frontend/__tests__/components/environment.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { shallow, mount } from 'enzyme';
+
+import { EnvironmentPage } from '../../public/components/environment';
+
+describe('EnvironmentPage', () => {
+
+  const objects={
+    kind: '',
+    apiVersion: '',
+    metadata: {},
+    spec: {},
+    status: {}
+  };
+
+  const environmentPageRO=<EnvironmentPage
+    obj={objects}
+    rawEnvData={[ { 'env': [ { 'name': 'test', 'value': ':0', 'ID': 0 } ] } ]}
+    envPath={[]}
+    readOnly={true}
+  />;
+
+  const environmentPage=<EnvironmentPage
+    obj={objects}
+    rawEnvData={[ { 'env': [ { 'name': 'test', 'value': ':0', 'ID': 0 } ] } ]}
+    envPath={['test']}
+    readOnly={false}
+  />;
+
+  it('does not show help text when page read-only', () => {
+    const wrapper=shallow(environmentPageRO);
+    expect(wrapper.find('p').exists()).toEqual(false);
+  });
+
```

Comment:
Throwing it out in case its helpful, if you have multiple states you want to test but don't want to do all the code by hand, you can always generate the tests with a loop:

```JavaScript 
const statuses = ['success', 'error', 'stale'];
statuses.forEach((status) => {
  describe(`When status attribute is set to ${status}`, () => {
    it(`Should render ${status} message`, () => {
       const wrapper=shallow(environmentPage);
       wrapper.setState({ status: status });
       expect(wrapper.find('.environment-buttons p').text()).toContain(status);
    });
  });
});
```
Though in this case I'm not sure if you can normalize across `success`, `errorMessage`, and `stale`. They are close, but a bit different properties.